source/dotnet/Library/AdaptiveCards/AdaptiveFallbackConverter.cs

```diff
@@ -0,0 +1,129 @@
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+
+namespace AdaptiveCards
+{
+    public class AdaptiveFallbackConverter : JsonConverter, ILogWarnings
+    {
+        public List<AdaptiveWarning> Warnings { get; set; } = new List<AdaptiveWarning>();
+
+        public override bool CanWrite => true;
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            AdaptiveFallbackElement fallback = (AdaptiveFallbackElement)value;
```

Comment:
This could potentially throw -- let;s just start using the non-throwing 'as' operator where we can for casting.